calculator.c

```diff
@@ -0,0 +1,147 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+
+int is_operator(char c)
+{
+    return c == '+' || c == '-' || c == '*' || c == '/';
+}
+
+int operator_precedence(char op)
+{
+    if (op == '*' || op == '/')
+        return 2;
+    if (op == '+' || op == '-')
+        return 1;
+    return 0;
+}
+
+float perform_operation(float a, float b, char op, int *error)
+{
+    switch (op)
+    {
+    case '+':
+        return a + b;
+    case '-':
+        return a - b;
+    case '*':
+        return a * b;
+    case '/':
+        if (b == 0)
+        {
+            *error = 1;
+            return 0;
+        }
+        return a / b;
+    default:
+        *error = 1;
+        return 0;

```

Comment:
remove multiple returns, use one return per method

Reply:
I have successfully removed multiple returns in one method